flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -516,7 +525,14 @@ public void handleOperatorEvent(OperatorEvent event) {
             checkWatermarkAlignment();
         } else if (event instanceof AddSplitEvent) {
             try {
-                sourceReader.addSplits(((AddSplitEvent<SplitT>) event).splits(splitSerializer));
+                List<SplitT> newSplits = ((AddSplitEvent<SplitT>) event).splits(splitSerializer);
```

Comment:
Is `handleOperatorEvent` executed by the mailbox thread as the other methods?

Reply:
Yes

https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java#L1426